action_server/frontend/src/views/actions/components/ActionRun.tsx

```diff
@@ -64,6 +77,64 @@ export const ActionRun: FC<Props> = ({ action, actionPackage }) => {
     [action, actionPackage, formData, apiKey, serverConfig],
   );
 
+  const onCodeChange = useCallback(
+    (code: string) => {
+      try {
+        setErrorJSON(undefined);
+        const rawInputData = JSON.parse(code as string);
+        const tempForm = [...formData];
+
+        Object.entries(rawInputData).forEach(([key, val]) => {
+          const foundFormItem = tempForm.find(
+            (elem) => elem.name.toLowerCase() === key.toLowerCase(),
+          );
+          if (foundFormItem) {
+            const foundFormItemIndex = tempForm.findIndex(
+              (elem) => elem.name.toLowerCase() === key.toLowerCase(),
+            );
+            tempForm[foundFormItemIndex] = { ...foundFormItem, value: val as string };
+          }
+        });
+
+        debounce(() => setFormData([...tempForm]), 750)();
+      } catch (e) {
+        setErrorJSON('Validating JSON syntax failed. Please verify the input and try again.');
+      }
+    },
+    [formData],
+  );
```

Comment:
This approach, would not work with nested items (which are flattened in the `formData`), which are supported by the spec.

Reply:
Seen that. Fixed it.